### DISCUSSION TITLE
To support my work at least say "Hello" here :)

### DISCUSSION THREAD

**Discussion by AdamGiergun** [AdamGiergun/adasss#1] [Announcements]
I've no idea if anybody is using this repo, so if you want to keep me motivated to update it on the regular basis at least just say hello here.
Also I'm not an expert in writing ebuilds, so, if you see something what might be done better, feel free to write here or propose change by making a [pull request](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/creating-a-pull-request).

**Comment by leycec** [comment]
I *thought* about using this repo – and thinking is half the battle. Pretty sure this guy said that.

![this guy, again](https://media2.giphy.com/media/lRRjGTRlFwmQYFmmpU/giphy_s.gif)

In short, I just wanted to extend my personal gratitude for your stalwart show of volunteerism. I also [maintain a poorly maintained Gentoo overlay](https://github.com/leycec/raiagent) – since literally before Gentoo even existed, possibly. Literally. So, I know how grinding and laborious overlay maintenance can be – which is why I do as little as possible.

Oh, and you should probably:

* Drop a link to the [official Gentoo Wiki article on Android Studio](https://wiki.gentoo.org/wiki/Android_studio), which is shockingly well-written and informative. In particular, users *will* need to scrupulously follow the [**"KVM and QEMU"** instructions to the letter](https://wiki.gentoo.org/wiki/Android_studio#KVM_and_QEMU). Android Studio is a fickle beast at best. Let's repeatedly whip the beast into submission and save everyone the pain of doing so without expert guidance.
* Update the installation instructions to note that your repository *must* be synced after being added: e.g.,

```bash
$ eselect repository enable adasss
$ emerge --sync adasss
```

Wouldn't it be nice of `eselect repository enable` did that for you? Of course, it doesn't. We can only ask for so much in 2022.

Thanks, Adam! You rock. :rock:

**Comment by AdamGiergun** [comment]
Thanks, @leycec  :grin:

For somebody using Gentoo for years, it's too easy to forget that some things might be not that obvious.

Regarding the article on the official Gentoo Wiki, it's a little shame for me to forget to mention it, especially when I contributed to it a few times :flushed:
However, I ought to disagree with the statement of the following "KVM and QEMU" section instructions to the letter. I don't have installed QEMU on my system and, nonetheless, the emulator still works :wink:

As for motives for maintaining this repository: due to the poor frequency of updates of official android-studio packages I was preparing my own packages for a long time (I'm using Android Studio on Gentoo on a daily basis), so it was simply a decision of sharing this work.

**Comment by leycec** [comment]
> I don't have installed QEMU on my system and, nonetheless, the emulator still works :wink:

**Fascinating...** *and horrifying.* I just fruitlessly wasted an evening getting my decrepit system back up to speed with modern QEMU, then. At least I feel a sense of pride and accomplishment. :woozy_face:

Thanks for nuthin', [official Gentoo Wiki article on Android Studio](https://wiki.gentoo.org/wiki/Android_studio).

> ...so it was simply a decision of sharing this work.

Thanks so much for making the right decision, Adam. If I had crypto, I would gift you abstract piles of it. Alas. Here we are. `</sigh>`

**Comment by AdamGiergun** [comment]
Pro tip: sometimes it's good to read not only a wiki page alone but also a discussion on it. :wink:

Regarding Gentoo, one of its very special bonuses is giving users satisfaction of solving very complex problems unknown to users of other distros :stuck_out_tongue:

**Comment by rahilarious** [comment]
This repo is great and all but I think your work deserves to be used by many more, it should be more discoverable. Giving back to community feels uniquely fulfilling. 

As you're already capable enough, it'll be easier for you to contribute into official repo by becoming Gentoo Dev, no more waiting on PR approvals. I definitely want to maintain some packages in future, but I'm barely scratching the surface on how to use git though.

**Comment by leycec** [comment]
> by becoming Gentoo Dev

*what could be simpler*

Seriously, though. If Adam wanted to do that, he'd have already done that. [We all know how to become Gentoo Devs](https://www.gentoo.org/get-involved/become-developer). Because we know, we know we don't want to become Gentoo Devs.

In my case, I enjoy the company of machines, cats, plants, and my wife – and that's it, really. Extroverted requirements like "participate in the community," "joining some of our IRC channels, subscribing to some of the mailing lists and/or using Gentoo forums," and "find a mentor" are a hard pass for most overlay maintainers. They're contrary to everything we find fun and decent in the world.

I doubt Adam's any different. But... maybe he is? I have been shocked and surprised before.

**Reply by rahilarious** [reply]
fair enough. So basically if you've LIFE, you can't be dev and add even more responsibility. Maybe my ambition to becoming dev comes from I'm denying social life and fun aspects not minding to stay on computer whole day/night.

**Reply by leycec** [reply]
**Yes.** This is it. Don't get us wrong, though: most of us also deny social life. We're just like you.

Also, Gentoo *is* the fun aspect. This is fun. It had better be! Right? Because nobody's gettin' paid. We're all volunteers. We do this for the joy of doing this. When we stop having fun, we stop doing this.

Lastly, it's not *just* that we're introverts. I've reflected on this... *and I've realized something new.* I'm mostly not a Gentoo dev because **I hate being told what to do.** When you're a Gentoo dev, many people tell you what to do on a daily basis – and you don't even get paid for it. This probably also explains why I live in a cabin in the Canadian wilderness with three cats and a beautiful wife.

## tl;dr
Do what you love. This is @leycec's Secret Road to Happiness in Only 42 Short Years. :smile_cat:

**Reply by rahilarious** [reply]
Gentoo users are used to reading long wikis and manuals, so we don't need tl;dr ;-)

**Comment by AdamGiergun** [comment]
@rahilarious 

leylec is right.
I know what it means to be a Gentoo developer, and I've done somehow similar things in the past. But I was much younger then. This would mean working with the computer for much longer than the typical 8 hours a day. It is impossible to continue this for a long time at my age. At my age, you'll see it clearly.

By the way, you wrote about GURU elsewhere. It is impossible to put packages that exist in the Gentoo root tree there, hence my decision to create this repository.
